File: qs-parser.js
const   url = require('url'),
        querystring = require('querystring');

[
    'http://foo.bar.com/?hello=jeff&hello=tom&hello=joe',
    'http://foo.bar.com/?hello=jeff,tom,joe',
    'http://foo.bar.com/?hello=jeff&tom&joe'
].forEach(function(rawUrl) {
    let parsedUrl = url.parse(rawUrl);
    let qs = querystring.parse(parsedUrl.query);
    console.log(rawUrl);
    console.log(JSON.stringify(qs));
});

File: qs-parser.py
from urllib.parse import urlparse

for rawUrl in [
    'http://foo.bar.com/?hello=jeff&hello=tom&hello=joe',
    'http://foo.bar.com/?hello=jeff,tom,joe',
    'http://foo.bar.com/?hello=jeff&tom&joe'
]: 
    print(rawUrl)
    o = urlparse(rawUrl)
    print(o.query)